src/main/java/seedu/address/model/person/Role.java

```diff
@@ -0,0 +1,152 @@
+     */
+    public Role(ModuleCode[] moduleCodes, String[] roleTypes) {
+        requireNonNull(moduleCodes);
+        requireNonNull(roleTypes);
+        if (moduleCodes.length != roleTypes.length) {
+            throw new IllegalArgumentException(
+                    "Module code array's length is not consistent with that of role type array.");
+        }
+        checkArgument(areValidRoleTypes(roleTypes), MESSAGE_CONSTRAINTS);
+        HashMap<ModuleCode, RoleType> newRoles = new HashMap<>();
+        for (int i = 0; i < moduleCodes.length; i++) {
+            newRoles.put(moduleCodes[i], roleTypeStringToEnum(roleTypes[i]));
+        }
+        this.roles = newRoles;
+    }
+
+    /**
+     * Constructor for a {@code Role}.
+     * Copy of the hashmap is performed to make the class immutable.
+     *
+     * @param roles new roles which are assigned to the Person
+     */
+    public Role(HashMap<ModuleCode, RoleType> roles) {
+        requireNonNull(roles);
+        this.roles = new HashMap<>(roles);
+    }
+
+    /**
+     * Returns true if the given role type String array contains all
+     * valid role type keywords.
+     *
+     * @param roleTypes Array of role type keywords
+     * @return whether the array contains all valid keywords
+     */
+    public static boolean areValidRoleTypes(String[] roleTypes) {
+        requireNonNull(roleTypes);
+        for (String keyword: roleTypes) {
+            if (!keyword.matches(VALIDATION_REGEX)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Converts a given role type keyword to its corresponding RoleType Enum value.
+     *
+     * @param keyword Keyword representing a role type.
+     * @return corresponding RoleType Enum value.
+     */
+    public static RoleType roleTypeStringToEnum(String keyword) {
+        requireNonNull(keyword);
+        switch (keyword.toLowerCase()) {
+        case "": // Fall through
+        case "student":
+            return RoleType.STUDENT;
+        case "tutor": // Fall through
+        case "ta":
+            return RoleType.TUTOR;
+        case "professor": // Fall through
+        case "prof":
+            return RoleType.PROFESSOR;
+        default:
+            throw new IllegalArgumentException("Invalid role type keyword for conversion to Enum value");
+        }
+    }
+
+    /**
+     * Get the String representation of a Person's Role for a specific
+     * role type.
+     *
+     * @param roleType Type of role to search and display.
+     * @return String representation of the roles with the specified type.
+     */
+    public String getRoleDescription(RoleType roleType) {
+        requireNonNull(roleType);
+        String prefix = roleType.toString() + " of: ";
+        StringBuilder description = new StringBuilder(prefix);
+        this.roles.entrySet().stream()
+                .filter(role -> role.getValue().equals(roleType))
+                .map(Map.Entry::getKey)
```

Comment:
do you want to extract this part out to a `private` function in this class? i think it might improve readability

then you can use `description.append(String.join(", ", list))` to avoid dealing with the trailing comma

Reply:
Thanks for the suggestion! I have extracted out this part of the logic and create two new methods `getFilteredModuleCodes(RoleType)` and `getModuleCodeString(Stream<ModuleCode>)` to improve the overall readability of the code and provide more useful APIs for future development 🎉